trustregistry/crud.py

```diff
@@ -51,26 +52,53 @@ def get_actor_by_id(db_session: Session, actor_id: str) -> db.Actor:
 
 def create_actor(db_session: Session, actor: Actor) -> db.Actor:
     bound_logger = logger.bind(body={"actor": actor})
-    bound_logger.info(
-        "Create actor in database. First assert actor ID does not already exist"
-    )
-    db_actor = db_session.query(db.Actor).filter(db.Actor.id == actor.id).one_or_none()
-
-    if db_actor:
-        bound_logger.info(
-            "Cannot create actor, as actor ID `{}` already exists in database.",
-            actor.id,
-        )
-        raise ActorAlreadyExistsException
-
-    bound_logger.debug("Adding actor to database")
-    db_actor = db.Actor(**actor.dict())
-    db_session.add(db_actor)
-    db_session.commit()
-    db_session.refresh(db_actor)
-
-    bound_logger.info("Successfully added actor to database.")
-    return db_actor
+    bound_logger.info("Try to create actor in database")
+
+    try:
+        bound_logger.debug("Adding actor to database")
+        db_actor = db.Actor(**actor.dict())
+        db_session.add(db_actor)
+        db_session.commit()
+        db_session.refresh(db_actor)
+
+        bound_logger.info("Successfully added actor to database.")
+        return db_actor
+
+    except IntegrityError as e:
+        db_session.rollback()
+        constraint_violation = str(e.orig).lower()
+
+        if f"actors.id" in constraint_violation:
+            bound_logger.info(
+                "Bad request: The requested actor ID already exists in database."
+            )
+            raise ActorAlreadyExistsException(
+                f"Bad request: The requested actor ID: `{actor.id}` already exists in database."
+            )
+
+        elif f"actors.name" in constraint_violation:
+            bound_logger.info(
+                "Bad request: The requested actor name already exists in database."
+            )
+            raise ActorAlreadyExistsException(
+                f"Bad request: The requested actor name: `{actor.name}` already exists in database."
+            )
+
+        elif f"actors.did" in constraint_violation:
+            if f"actors.didcomm_invitation" in constraint_violation:
+                bound_logger.info(
+                    "Bad request: The requested actor DIDComm invitation already exists in database."
+                )
+                raise ActorAlreadyExistsException(
+                    f"Bad request: The requested actor DIDComm: `{actor.didcomm_invitation}` invitation already exists in database."
```

Comment:
Here there's a missing "invitation" string in the message.

Also - this didcomm_invitation can be a HUGE string. I think it's fair to exclude it here, and just say: "An actor with this didcomm invitation already exists."

I think each of the "The requested actor ... already exists" strings can rather just say "An actor with ...  already exists"